src/t_set.c

```diff
@@ -353,10 +355,8 @@ void setTypeReleaseIterator(setTypeIterator *si) {
  *
  * When there are no more elements -1 is returned. */
 int setTypeNext(setTypeIterator *si, char **str, size_t *len, int64_t *llele) {
-    if (si->encoding == OBJ_ENCODING_HT) {
-        dictEntry *de = dictNext(si->di);
-        if (de == NULL) return -1;
-        *str = dictGetKey(de);
+    if (si->encoding == OBJ_ENCODING_HASHTABLE) {
+        if (!hashtableNext(si->hashtable_iterator, (void **)str)) return -1;
         *len = sdslen(*str);
```

Comment:
How can we avoid the cast here? Like this?

```suggestion
        void *next;
        if (!hashtableNext(si->hashtable_iterator, &next)) return -1;
        *str = next;
        *len = sdslen(next);
```